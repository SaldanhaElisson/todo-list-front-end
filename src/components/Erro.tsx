import React from "react";
import { Box, Typography, Button } from "@mui/material";
import ErrorOutlineIcon from "@mui/icons-material/ErrorOutline";

interface ErroProps {
  message: string;
  showBackButton?: boolean;
}

const Erro: React.FC<ErroProps> = ({ message, showBackButton = false }) => {
  return (
    <Box
      sx={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        height: "100%",
      }}
    >
      <Box
        sx={{
          backgroundColor: "error.main",
          color: "white",
          padding: "2rem",
          borderRadius: "8px",
          boxShadow: "0 4px 8px rgba(0, 0, 0, 0.1)",
          textAlign: "center",
        }}
      >
        <ErrorOutlineIcon sx={{ fontSize: 40, marginBottom: "1rem" }} />
        <Typography variant="h6" gutterBottom>
          Ocorreu um erro
        </Typography>
        <Typography variant="body1">{message}</Typography>
        {showBackButton && (
          <Button
            variant="contained"
            color="secondary"
            sx={{ marginTop: "1rem" }}
            onClick={() => window.history.back()}
          >
            Voltar
          </Button>
        )}
      </Box>
    </Box>
  );
};

export default Erro;
