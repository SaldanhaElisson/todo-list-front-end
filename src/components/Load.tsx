import { Box, CircularProgress } from "@mui/material";
import React from "react";

export const Load: React.FC = () => {
  return (
    <Box
      sx={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        height: "30vh",
      }}
      data-testid="box-container"
    >
      <CircularProgress />
    </Box>
  );
};
