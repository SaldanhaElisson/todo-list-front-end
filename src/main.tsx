import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import Routers from "./Routers.tsx";
import "./index.css";
import { BrowserRouter } from "react-router-dom";

createRoot(document.getElementById("root")!).render(
  <StrictMode>
    <BrowserRouter>
      <Routers />
    </BrowserRouter>
  </StrictMode>
);
