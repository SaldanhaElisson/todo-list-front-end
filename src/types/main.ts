export interface ITodo {
  id: string;
  title: string;
  description: string;
  date: string;
  completed: boolean;
}
