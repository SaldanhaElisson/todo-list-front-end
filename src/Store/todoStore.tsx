import { create } from "zustand";
import { ITodo } from "../types/main";

interface TodoState {
  addTodoStore: (newTodos: ITodo[]) => void;
  todos: ITodo[];
}

export const useTodoStore = create<TodoState>((set) => ({
  todos: [],
  addTodoStore: (newTodos: ITodo[]) => set(() => ({ todos: newTodos })),
}));
