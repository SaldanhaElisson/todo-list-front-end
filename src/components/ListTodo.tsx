import {
  Checkbox,
  List,
  ListItem,
  ListItemText,
  Typography,
} from "@mui/material";
import React from "react";
import { Link } from "react-router-dom";
import { ITodo } from "../types/main";
import dayjs from "dayjs";

interface Iprops {
  todos: Array<ITodo>;
  handleToggleComplete: (index: number) => void;
}

export const ListTodo: React.FC<Iprops> = ({ todos, handleToggleComplete }) => {
  // Função auxiliar para verificar se a data do todo já passou
  const isDatePassed = (date: string): boolean => {
    return dayjs(date).isBefore(dayjs(), "day");
  };

  return (
    <List
      sx={{
        width: "100%",
      }}
      data-testid="lista-todo"
    >
      {todos.map((todo, index) => (
        <ListItem
          key={todo.id}
          sx={{
            marginBottom: "1rem",
            backgroundColor: todo.completed
              ? "lightgray"
              : isDatePassed(todo.date)
              ? "#F05D5E"
              : "#D7EC98",
            borderRadius: "5px",
          }}
          aria-labelledby={`todo-title-${todo.id}`}
        >
          <Checkbox
            edge="start"
            checked={!!todo.completed}
            onChange={() => handleToggleComplete(index)}
            tabIndex={-1}
            disableRipple
            inputProps={{
              "aria-labelledby": `todo-title-${todo.id}`,
            }}
          />
          <Link
            to={`/todo/${todo.id}`}
            style={{ textDecoration: "none", color: "inherit" }}
            aria-label={`Ver detalhes da tarefa ${todo.title}`}
          >
            <ListItemText
              primary={
                <Typography
                  variant="h6"
                  component="div"
                  color={"textSecondary"}
                  id={`todo-title-${todo.id}`}
                >
                  {todo.title}
                </Typography>
              }
              secondary={
                <>
                  <Typography variant="body1" component="span">
                    {todo.description}
                  </Typography>
                  <br />
                  <Typography variant="body2" component="span">
                    {todo.date}
                  </Typography>
                </>
              }
              sx={{
                textDecoration: todo.completed ? "line-through" : "none",
                cursor: "pointer",
              }}
            />
          </Link>
        </ListItem>
      ))}
    </List>
  );
};
