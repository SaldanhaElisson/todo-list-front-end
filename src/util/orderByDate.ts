import { ITodo } from "../types/main";

export function organizeTodosByDate(todos: ITodo[]): ITodo[] {
  return todos.sort(
    (a, b) => new Date(a.date).getTime() - new Date(b.date).getTime()
  );
}
