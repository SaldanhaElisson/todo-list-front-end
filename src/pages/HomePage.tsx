import React, { useState } from "react";
import { Box, TextField, Button } from "@mui/material";
import { Date } from "../components/Date";
import AddIcon from "@mui/icons-material/Add";
import { ITodo } from "../types/main";
import { ListTodo } from "../components/ListTodo";
import dayjs, { Dayjs } from "dayjs";
import { useFetchTodos } from "../hook/useFetchTodos";
import { todoSchema } from "../util/schemas";
import Erro from "../components/Erro";
import { organizeTodosByDate } from "../util/orderByDate";
import { Load } from "../components/Load";

export const HomePage: React.FC = () => {
  const { todos, setTodos, loading, error, addTodo, updateTodo } =
    useFetchTodos();

  const [title, setTitle] = useState<string>("");
  const [description, setDescription] = useState<string>("");
  const [date, setDate] = useState<Dayjs>(dayjs());
  const [validationError, setValidationError] = useState<{
    field: string;
    message: string;
  } | null>(null);
  const [timeoutId, setTimeoutId] = useState<NodeJS.Timeout | null>(null);
  const [pendingUpdates, setPendingUpdates] = useState<ITodo[]>([]);
  console.log(todos);

  const handleToggleComplete = (index: number) => {
    // Atualiza o estado local imediatamente
    setTodos((prevTodos) =>
      prevTodos.map((todo, i) =>
        i === index ? { ...todo, completed: !todo.completed } : todo
      )
    );

    // Adiciona a tarefa atualizada ao estado de atualizações pendentes
    setPendingUpdates((prevUpdates) => {
      const updatedTodo = {
        ...todos[index],
        completed: !todos[index].completed,
      };
      return [
        ...prevUpdates.filter((todo) => todo.id !== updatedTodo.id),
        updatedTodo,
      ];
    });

    // Limpa o timeout anterior, se existir
    if (timeoutId) {
      clearTimeout(timeoutId);
    }

    // Define um novo timeout para salvar no banco de dados após 5 segundos
    const newTimeoutId = setTimeout(async () => {
      // Salva todas as atualizações pendentes no banco de dados
      setPendingUpdates((prevUpdates) => {
        prevUpdates.forEach(async (todo) => {
          console.log("fiz update" + todo);
          await updateTodo(todo.id ?? "", todo);
        });
        return [];
      });
    }, 5000);

    setTimeoutId(newTimeoutId);
  };

  const handleAddTodo = async (): Promise<void> => {
    const newTodo: Omit<ITodo, "id"> = {
      title,
      description,
      date: date ? date.format("YYYY-MM-DD") : "",
      completed: false,
    };

    const validation = todoSchema.safeParse(newTodo);

    if (!validation.success) {
      const error = validation.error.errors[0];

      setValidationError({
        field: error.path[0] as string,
        message: error.message,
      });
      return;
    }

    await addTodo(newTodo);
    setTitle("");
    setDescription("");
    setDate(dayjs("2022-04-17"));
    setValidationError(null);
  };

  if (error && error.includes("Erro no servidor")) {
    return <Erro message={error} />;
  }

  // Organizar os todos por data antes de renderizar
  const sortedTodos = organizeTodosByDate(todos);

  return (
    <>
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          padding: "1rem",
          "@media (min-width: 600px)": {
            padding: "2rem",
          },
        }}
        data-testid="home-page"
      >
        <Box
          sx={{
            display: "flex",
            flexDirection: "column",
            marginBottom: "1rem",
            "@media (min-width: 600px)": {
              flexDirection: "row",
              justifyContent: "space-between",
              alignItems: "center",
            },
          }}
          data-testid="input-container"
        >
          <TextField
            label="Título"
            fullWidth
            variant="outlined"
            sx={{
              marginBottom: "1rem",
              "@media (min-width: 600px)": {
                marginBottom: "0",
                marginRight: "1rem",
              },
            }}
            value={title}
            onChange={(e) => setTitle(e.target.value)}
            aria-label="Título"
            error={validationError?.field === "title"}
            helperText={
              validationError?.field === "title" ? validationError.message : ""
            }
            data-testid="title-input"
          />
          <Date setValue={setDate} value={date} />
        </Box>
        <TextField
          label="Descrição"
          fullWidth
          variant="outlined"
          sx={{
            marginBottom: "1rem",
          }}
          value={description}
          onChange={(e) => setDescription(e.target.value)}
          aria-label="Descrição"
          error={validationError?.field === "description"}
          helperText={
            validationError?.field === "description"
              ? validationError.message
              : ""
          }
          data-testid="description-input"
        />

        <Box
          sx={{
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            width: "100%",
          }}
          data-testid="button-container"
        >
          <Button
            variant="contained"
            color="primary"
            sx={{
              marginY: "1rem",
              maxWidth: "12.5rem",
            }}
            startIcon={<AddIcon />}
            aria-label="Adicionar Tarefa"
            onClick={handleAddTodo}
            disabled={loading}
            data-testid="add-button"
          >
            Adicionar
          </Button>
        </Box>

        {loading && <Load data-testid="loading" />}
        {error && <p data-testid="erro">{error}</p>}
        {todos.length !== 0 && (
          <ListTodo
            todos={sortedTodos}
            handleToggleComplete={handleToggleComplete}
          />
        )}
      </Box>
    </>
  );
};
