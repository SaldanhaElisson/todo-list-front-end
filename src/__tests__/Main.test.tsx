import { render, screen, fireEvent, renderHook } from "@testing-library/react";
import "@testing-library/jest-dom";
import { describe, it, expect, beforeEach, vi } from "vitest";
import { HomePage } from "../pages/HomePage";
import dayjs from "dayjs";
import { MemoryRouter } from "react-router-dom";
import { ListTodo } from "../components/ListTodo";
import { Load } from "../components/Load";
import Erro from "../components/Erro";
import { useTodoStore } from "../Store/todoStore";
import { ITodo } from "../types/main";
import { act } from "react";
vi.mock("../hook/useFetchTodos");
import { organizeTodosByDate } from "../util/orderByDate";

beforeEach(() => {
  vi.spyOn(window.history, "back").mockImplementation(() => {});
  vi.mock("../hook/useFetchTodos", () => ({
    useFetchTodos: () => ({
      todos: [
        {
          id: "1",
          title: "Nova Tarefa",
          description: "Descrição da nova tarefa",
          date: dayjs().format("YYYY-MM-DD"),
          completed: false,
        },
      ],
      setTodos: vi.fn(),
      loading: false,
      error: null,
      addTodo: vi.fn(),
      updateTodo: vi.fn(),
      getTodoById: vi.fn().mockReturnValue({
        id: "1",
        title: "Nova Tarefa",
        description: "Descrição da nova tarefa",
        date: dayjs().format("YYYY-MM-DD"),
      }),
    }),
  }));
});

describe("HomePage", () => {
  it("deve renderizar a HomePage corretamente", () => {
    const { container } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("verificar se os campos estão na página e podem ser alterados", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    const titleInput = screen
      .getByTestId("title-input")
      ?.querySelector("input");
    const descriptionInput = screen
      .getByTestId("description-input")
      ?.querySelector("input");

    // Verificar e alterar o campo de título
    fireEvent.change(titleInput, { target: { value: "Novo Título" } });
    expect(titleInput.value).toBe("Novo Título");
  });
});

const mockTodos = [
  {
    id: "1",
    title: "Tarefa 1",
    description: "Descrição da Tarefa 1",
    date: dayjs().format("YYYY-MM-DD"),
    completed: false,
  },
  {
    id: "2",
    title: "Tarefa 2",
    description: "Descrição da Tarefa 2",
    date: dayjs().subtract(1, "day").format("YYYY-MM-DD"), // Data passada
    completed: true,
  },
];

const handleToggleComplete = vi.fn();

describe("ListTodo", () => {
  it("deve renderizar a lista de todos corretamente", () => {
    render(
      <MemoryRouter>
        <ListTodo
          todos={mockTodos}
          handleToggleComplete={handleToggleComplete}
        />
      </MemoryRouter>
    );

    // Verificar se os itens da lista estão presentes
    const todoItems = screen.getAllByRole("listitem");
    expect(todoItems).toHaveLength(mockTodos.length);

    // Verificar se os títulos estão corretos
    expect(screen.getByText("Tarefa 1")).toBeInTheDocument();
    expect(screen.getByText("Tarefa 2")).toBeInTheDocument();

    // Verificar se as descrições estão corretas
    expect(screen.getByText("Descrição da Tarefa 1")).toBeInTheDocument();
    expect(screen.getByText("Descrição da Tarefa 2")).toBeInTheDocument();

    // Verificar se as datas estão corretas
    expect(screen.getByText(dayjs().format("YYYY-MM-DD"))).toBeInTheDocument();
    expect(
      screen.getByText(dayjs().subtract(1, "day").format("YYYY-MM-DD"))
    ).toBeInTheDocument();
  });

  it("deve chamar handleToggleComplete ao clicar no checkbox", () => {
    render(
      <MemoryRouter>
        <ListTodo
          todos={mockTodos}
          handleToggleComplete={handleToggleComplete}
        />
      </MemoryRouter>
    );

    const checkboxes = screen.getAllByRole("checkbox");
    fireEvent.click(checkboxes[0]);

    expect(handleToggleComplete).toHaveBeenCalledWith(0);
  });
});

describe("Load", () => {
  it("deve renderizar o componente Load corretamente", () => {
    render(<Load />);

    // Verificar se o CircularProgress está presente
    const circularProgress = screen.getByRole("progressbar");
    expect(circularProgress).toBeInTheDocument();

    // Verificar se o Box está presente
    const box = screen.getByTestId("box-container");
    expect(box).toBeInTheDocument();
  });
});

describe("Erro", () => {
  it("deve renderizar a mensagem de erro corretamente", () => {
    const message = "Este é um erro de teste";
    render(<Erro message={message} />);

    // Verificar se a mensagem de erro está presente
    expect(screen.getByText("Ocorreu um erro")).toBeInTheDocument();
    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it("deve renderizar o botão de voltar quando showBackButton é true", () => {
    const message = "Este é um erro de teste";
    render(<Erro message={message} showBackButton={true} />);

    // Verificar se o botão de voltar está presente
    const backButton = screen.getByRole("button", { name: /voltar/i });
    expect(backButton).toBeInTheDocument();

    // Simular clique no botão de voltar
    fireEvent.click(backButton);
    // Verificar se a função de voltar foi chamada
    expect(window.history.back).toHaveBeenCalled();
  });

  it("não deve renderizar o botão de voltar quando showBackButton é false", () => {
    const message = "Este é um erro de teste";
    render(<Erro message={message} showBackButton={false} />);

    // Verificar se o botão de voltar não está presente
    const backButton = screen.queryByRole("button", { name: /voltar/i });
    expect(backButton).not.toBeInTheDocument();
  });
});

describe("useTodoStore", () => {
  it("should add new todos to the store", () => {
    const { result } = renderHook(() => useTodoStore());

    const newTodos: ITodo[] = [
      {
        id: "1",
        title: "Nova Tarefa",
        description: "Descrição da nova tarefa",
        date: "2024-08-14",
        completed: false,
      },
    ];

    act(() => {
      result.current.addTodoStore(newTodos);
    });

    expect(result.current.todos).toEqual(newTodos);
  });
});

describe("organizeTodosByDate", () => {
  it("deve organizar os todos por data em ordem ascendente", () => {
    const todos: ITodo[] = [
      {
        id: "1",
        title: "Tarefa 1",
        description: "Descrição da Tarefa 1",
        date: "2024-08-15",
        completed: false,
      },
      {
        id: "2",
        title: "Tarefa 2",
        description: "Descrição da Tarefa 2",
        date: "2024-08-14",
        completed: true,
      },
      {
        id: "3",
        title: "Tarefa 3",
        description: "Descrição da Tarefa 3",
        date: "2024-08-16",
        completed: false,
      },
    ];

    const sortedTodos = organizeTodosByDate(todos);

    expect(sortedTodos[0].date).toBe("2024-08-14");
    expect(sortedTodos[1].date).toBe("2024-08-15");
    expect(sortedTodos[2].date).toBe("2024-08-16");
  });
});
