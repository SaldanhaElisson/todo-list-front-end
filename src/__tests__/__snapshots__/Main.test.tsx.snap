// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > deve renderizar a HomePage corretamente 1`] = `
<div>
  <div
    class="MuiBox-root css-1dyth91"
    data-testid="home-page"
  >
    <div
      class="MuiBox-root css-rzt50z"
      data-testid="input-container"
    >
      <div
        aria-label="Título"
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1jssvtq-MuiFormControl-root-MuiTextField-root"
        data-testid="title-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Título
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Título
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r1:"
          id=":r1:-label"
        >
          Data de Vencimento
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            inputmode="text"
            placeholder="MM/DD/YYYY"
            type="text"
            value="08/14/2024"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date, selected date is Aug 14, 2024"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Data de Vencimento
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      aria-label="Descrição"
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1jknhqr-MuiFormControl-root-MuiTextField-root"
      data-testid="description-input"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r3:"
        id=":r3:-label"
      >
        Descrição
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r3:"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Descrição
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiBox-root css-1pxgmwj"
      data-testid="button-container"
    >
      <button
        aria-label="Adicionar Tarefa"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1w39yky-MuiButtonBase-root-MuiButton-root"
        data-testid="add-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
            />
          </svg>
        </span>
        Adicionar
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      class="MuiList-root MuiList-padding css-1lbrbe2-MuiList-root"
      data-testid="lista-todo"
    >
      <li
        aria-labelledby="todo-title-1"
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1xjwxd1-MuiListItem-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-edgeStart MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            aria-labelledby="todo-title-1"
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            tabindex="-1"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <a
          aria-label="Ver detalhes da tarefa Nova Tarefa"
          href="/todo/1"
          style="text-decoration: none;"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-3ypqcr-MuiListItemText-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
              id="todo-title-1"
            >
              Nova Tarefa
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Descrição da nova tarefa
              </span>
              <br />
              <span
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                2024-08-14
              </span>
            </p>
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
